apps/src/templates/MultipleSectionsAssigner.jsx

```diff
@@ -0,0 +1,350 @@
+import {
+  assignToSection,
+  testingFunction,
+  unassignSection
+} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+import {updateHiddenScript} from '@cdo/apps/code-studio/hiddenLessonRedux';
+
+class MultipleSectionsAssigner extends Component {
+  static propTypes = {
+    courseId: PropTypes.number,
+    assignmentName: PropTypes.string.isRequired,
+    onClose: PropTypes.func.isRequired,
+    forceReload: PropTypes.bool,
+    courseOfferingId: PropTypes.number,
+    courseVersionId: PropTypes.number,
+    scriptId: PropTypes.number,
+    reassignConfirm: PropTypes.func,
+    isOnCoursePage: PropTypes.bool,
+    isStandAloneUnit: PropTypes.bool,
+    participantAudience: PropTypes.string,
+    // Redux
+    sections: PropTypes.arrayOf(sectionForDropdownShape).isRequired,
+    unassignSection: PropTypes.func.isRequired,
+    assignToSection: PropTypes.func.isRequired,
+    updateHiddenScript: PropTypes.func.isRequired,
+    testingFunction: PropTypes.func.isRequired,
+    selectedSectionId: PropTypes.number
+  };
+
+  constructor(props) {
+    super(props);
+
+    let initialSectionsAssignedToCourseList = [];
+
+    // check to see if this is coming from the UNIT landing page - if so add courses featuring this unit
+    if (!this.props.isOnCoursePage) {
+      if (this.props.isStandAloneUnit) {
+        for (let i = 0; i < this.props.sections.length; i++) {
+          if (
+            this.props.courseVersionId ===
+            this.props.sections[i].courseVersionId
+          ) {
+            initialSectionsAssignedToCourseList.push(this.props.sections[i]);
+          }
+        }
+      } else {
+        for (let i = 0; i < this.props.sections.length; i++) {
+          if (this.props.scriptId === this.props.sections[i].unitId) {
+            initialSectionsAssignedToCourseList.push(this.props.sections[i]);
+          }
+        }
+      }
+    } else if (this.props.isOnCoursePage) {
+      // checks to see if this is coming from the COURSE landing page
+      for (let i = 0; i < this.props.sections.length; i++) {
+        if (this.props.courseId === this.props.sections[i].courseId) {
+          initialSectionsAssignedToCourseList.push(this.props.sections[i]);
+        }
+      }
+    }
+
+    this.state = {
+      currentSectionsAssigned: initialSectionsAssignedToCourseList,
+      initialSectionsAssigned: initialSectionsAssignedToCourseList // I am wondering if this should be passed in through props bc it never changes.
+    };
+  }
+
+  chooseMenuItem = section => {
+    console.log(section + 'was clicked');
+  };
+
+  handleChangedCheckbox = currentSection => {
+    const isUnchecked = !!this.state.currentSectionsAssigned.some(
+      s => s.code === currentSection.code
+    );
+    if (isUnchecked) {
+      this.setState(state => {
+        let newList = state.currentSectionsAssigned.filter(
+          s => s.code !== currentSection.code
+        );
+        return {currentSectionsAssigned: newList};
```

Comment:
it looks like this function is used as an onclick handler -- why do you need to return something?

Reply:
sorry I misread the code! I think this is fine, though normally I would just extract this logic out and do something like:

```
const {currentSectionsAssigned} = this.state;
const newList = ....;
this.setState({currrentSectionsAssign: newList});
```

I don't know that one way is better than the other, so I think this is fine as is!